tests/test_operational.c

```diff
@@ -5071,6 +5076,343 @@ test_state_default_merge(void **state)
+    ret = pthread_barrier_init(&st->barrier, NULL, 5);
+    assert_int_equal(ret, 0);
+
+    /* subscribe as state data provider */
+    ret = sr_oper_get_subscribe(st->sess, "ietf-interfaces", "/ietf-interfaces:interfaces-state", parallel_shm_access_cb,
+            *state, 0, &subscr1);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    /* subscribe as state data provider */
+    ret = sr_oper_get_subscribe(st->sess, "ietf-interfaces", "/ietf-interfaces:interfaces-state", parallel_shm_access_cb,
+            *state, SR_SUBSCR_OPER_MERGE, &subscr2);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    /* subscribe as state data provider */
+    ret = sr_oper_get_subscribe(st->sess, "ietf-interfaces", "/ietf-interfaces:interfaces-state", parallel_shm_access_cb,
+            *state, SR_SUBSCR_OPER_MERGE, &subscr3);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    /* subscribe as state data provider */
+    ret = sr_oper_get_subscribe(st->sess, "ietf-interfaces", "/ietf-interfaces:interfaces-state", parallel_shm_access_cb,
+            *state, SR_SUBSCR_OPER_MERGE, &subscr4);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    /* subscribe as state data provider */
+    ret = sr_oper_get_subscribe(st->sess, "ietf-interfaces", "/ietf-interfaces:interfaces-state", parallel_shm_access_cb,
+            *state, SR_SUBSCR_OPER_MERGE, &subscr5);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    /* try to read them back from operational DS */
+    ret = sr_session_switch_ds(st->sess, SR_DS_OPERATIONAL);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    /* read all data from operational again */
+    ret = sr_get_data(st->sess, "/ietf-interfaces:*", 0, 0, SR_OPER_WITH_ORIGIN, &data);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    sr_release_data(data);
+
+    sr_unsubscribe(subscr1);
+    sr_unsubscribe(subscr2);
+    sr_unsubscribe(subscr3);
+    sr_unsubscribe(subscr4);
+    sr_unsubscribe(subscr5);
+
+    /* see if EBUSY is thrown */
+    ret = pthread_barrier_destroy(&st->barrier);
+    assert_int_not_equal(ret, EBUSY);
+    assert_int_equal(ret, 0);
+}
+
+static int
+callback_success_cb(sr_session_ctx_t *session, uint32_t sub_id, const char *module_name, const char *xpath,
+        const char *request_xpath, uint32_t request_id, struct lyd_node **parent, void *private_data)
+{
+    (void)session;
+    (void)sub_id;
+    (void)module_name;
+    (void)xpath;
+    (void)request_xpath;
+    (void)request_id;
+    (void)parent;
+    (void)private_data;
+
+    return SR_ERR_OK;
+}
+
+static int
+callback_fail_cb(sr_session_ctx_t *session, uint32_t sub_id, const char *module_name, const char *xpath,
+        const char *request_xpath, uint32_t request_id, struct lyd_node **parent, void *private_data)
+{
+    (void)session;
+    (void)sub_id;
+    (void)module_name;
+    (void)xpath;
+    (void)request_xpath;
+    (void)request_id;
+    (void)parent;
+    (void)private_data;
+
+    return SR_ERR_CALLBACK_FAILED;
+}
+
+static void
+test_callback_failed(void **state)
```

Comment:
You should be able to use barriers in this test as well.

Reply:
Done